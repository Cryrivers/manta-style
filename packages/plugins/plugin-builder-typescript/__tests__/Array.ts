import getTranspiledString from '../src/utils/transpile';

describe('Array', () => {
  test('Bracket form', () => {
    expect(getTranspiledString('type Test = string[]')).toMatchSnapshot();
    expect(
      getTranspiledString('type Test = (number | string)[]'),
    ).toMatchSnapshot();
  });
  test('Array keyword form', () => {
    expect(getTranspiledString('type Test = Array<string>;')).toMatchSnapshot();
    expect(
      getTranspiledString('type Test = Array<number | string>;'),
    ).toMatchSnapshot();
    expect(getTranspiledString('type Test<T> = Array<T>;')).toMatchSnapshot();
  });
});
