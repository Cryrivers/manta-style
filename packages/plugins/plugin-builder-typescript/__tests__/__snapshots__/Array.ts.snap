// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Array Array keyword form 1`] = `
"\\"use strict\\";
exports.__esModule = true;
var runtime_1 = require(\\"@manta-style/runtime\\");
var Test = runtime_1[\\"default\\"].TypeAliasDeclaration(\\"Test\\", function (typeFactory) {
    var type = runtime_1[\\"default\\"].ArrayType(runtime_1[\\"default\\"].StringKeyword);
    return type;
}, []);
"
`;

exports[`Array Array keyword form 2`] = `
"\\"use strict\\";
exports.__esModule = true;
var runtime_1 = require(\\"@manta-style/runtime\\");
var Test = runtime_1[\\"default\\"].TypeAliasDeclaration(\\"Test\\", function (typeFactory) {
    var type = runtime_1[\\"default\\"].ArrayType(runtime_1[\\"default\\"].ParenthesizedType(runtime_1[\\"default\\"].UnionType([runtime_1[\\"default\\"].NumberKeyword, runtime_1[\\"default\\"].StringKeyword])));
    return type;
}, []);
"
`;

exports[`Array Array keyword form 3`] = `
"\\"use strict\\";
exports.__esModule = true;
var runtime_1 = require(\\"@manta-style/runtime\\");
var Test = runtime_1[\\"default\\"].TypeAliasDeclaration(\\"Test\\", function (typeFactory) {
    var T = typeFactory.TypeParameter(\\"T\\");
    var type = runtime_1[\\"default\\"].ArrayType(T);
    return type;
}, []);
"
`;

exports[`Array Bracket form 1`] = `
"\\"use strict\\";
exports.__esModule = true;
var runtime_1 = require(\\"@manta-style/runtime\\");
var Test = runtime_1[\\"default\\"].TypeAliasDeclaration(\\"Test\\", function (typeFactory) {
    var type = runtime_1[\\"default\\"].ArrayType(runtime_1[\\"default\\"].StringKeyword);
    return type;
}, []);
"
`;

exports[`Array Bracket form 2`] = `
"\\"use strict\\";
exports.__esModule = true;
var runtime_1 = require(\\"@manta-style/runtime\\");
var Test = runtime_1[\\"default\\"].TypeAliasDeclaration(\\"Test\\", function (typeFactory) {
    var type = runtime_1[\\"default\\"].ArrayType(runtime_1[\\"default\\"].ParenthesizedType(runtime_1[\\"default\\"].UnionType([runtime_1[\\"default\\"].NumberKeyword, runtime_1[\\"default\\"].StringKeyword])));
    return type;
}, []);
"
`;
